core/Http.php

```diff
@@ -830,4 +827,28 @@ public static function getModifiedSinceHeader()
         }
         return $modifiedSince;
     }
+
+    /**
+     * Returns Proxy to use for connecting via HTTP to given URL
+     *
+     * @param string $url
+     * @return array
+     */
+    private static function getProxyConfiguration($url)
+    {
+        $hostname = UrlHelper::getHostFromUrl($url);
+        $localHostnames = Url::getLocalHostnames();
+
+        if(in_array($hostname, $localHostnames)) {
+            return array(null, null, null, null);
+        }
+
+        // proxy configuration
+        $proxyHost = Config::getInstance()->proxy['host'];
+        $proxyPort = Config::getInstance()->proxy['port'];
+        $proxyUser = Config::getInstance()->proxy['username'];
+        $proxyPassword = Config::getInstance()->proxy['password'];
+
+        return array($proxyHost, $proxyPort, $proxyUser, $proxyPassword);
```

Comment:
Just a notice: may be it's better to use associative array or something like `ProxyConfig` class with public methods, returning `null` when it's local? Also I don't understand why this method is `private`, not protected?